bin/varnishd/cache/cache_conn_pool.c

```diff
@@ -249,17 +259,39 @@ VCP_Rel(struct conn_pool **cpp)
 		(void)shutdown(pfd->fd, SHUT_RDWR);
 		cp->n_kill++;
 	}
-	while (cp->n_kill) {
-		Lck_Unlock(&cp->mtx);
-		(void)usleep(20000);
+
+	ALLOC_OBJ(dcp, DYING_CP_MAGIC);
+	dcp->cp = cp;
+	VTAILQ_INSERT_TAIL(&dying_cps, dcp, list);
+	Lck_Unlock(&cp->mtx);
```

Comment:
You shouldn't need an extra data structure for this, `cp` should carry the `list` field. It could be made more explicit, for example `kill_list` (a new `VTAILQ_ENTRY` in `struct conn_pool`).

Reply:
I see we `rushed` our replies.
I think he could reuse the VBRT entry instead. No need to add a new member.

Reply:
That is another option, indeed.